### DISCUSSION TITLE
Modifying text does not reset the origin but continues scrolling.

### DISCUSSION THREAD

**Discussion by HOLICCC** [cbpowell/MarqueeLabel#290] [Q&A] [answered]
### I'd like to not reset the Marquee Label back to the starting point after modifying the text, but I can't do it.
+ OS: iOS
+ Language: Objective-C
--- 
+ Expect that when changes are made to the text attribute of a MarqueeLabel, the position is not repositioned to the origin.
+ In fact, it is not possible to modify the MarqueeLabel and continue scrolling, instead of resetting to the origin in scrolling.

**Comment by cbpowell** [comment, marked as answer]
Hi! Unfortunately that’s not currently possibly with the default functionality.

Part of the reason I never implemented it is because it is because the scroll animation timing and positioning is inherently dependent on the width/length of the label, which is a function of the label string. That means modifying in-flight animations and covering edge cases, and I’ve never had the time to take that on.

It’s not *impossible* to make it work though I don’t think, but it would probably take some changes to make it work well!

**Reply by HOLICCC** [reply]
All right~
Thanks.
